generators/parser.go

```diff
@@ -0,0 +1,58 @@
+package generators
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type KindGroup struct {
+	Kind string `json:"kind"`
+}
+type ApiVersionGroup struct {
+	ApiVersion string `json:"apiVersion"`
+}
+
+type MetadataAttri struct {
+	Name string `json:"name"`
+}
+type MetadataGroup struct {
+	Name MetadataAttri `json:"metadata,inline"`
+}
+
+type Resources struct {
+	Size string `json:"storage"`
+}
+
+type Spec struct {
+	AccessModes []string `json:"accessModes"`
+	Resources   `json:"resources"`
+}
+
+type SpecGroup struct {
+	Spec `json:"spec"`
+}
+
+type PVC struct {
+	KindGroup       `json:",inline"`
+	ApiVersionGroup `json:",inline"`
+	MetadataGroup   `json:",inline"`
+	SpecGroup       `json:",inline"`
+}
+
+func Parse(r io.Reader) (PVC, error) {
+	fmt.Println("Yaml Parser Demo:")
+
+	reader := yaml.NewYAMLToJSONDecoder(r)
+	res := PVC{}
+	err := reader.Decode(&res)
+	if err != nil {
+		fmt.Errorf("%v", err)
```

Comment:
You can remove this. This only formats an error, it doesn't actually print anything out: https://pkg.go.dev/fmt?tab=doc#Errorf